app/src/ui/lib/sandboxed-markdown.tsx

```diff
@@ -0,0 +1,165 @@
+  }
+
+  /**
+   * Since iframe styles are isolated from the rest of the app, we have a
+   * markdown.css file that we added to app/static directory that we can read in
+   * and provide to the iframe.
+   *
+   * Additionally, the iframe will not be aware of light/dark theme variables,
+   * thus we will scrape the subset of them needed for the markdown css from the
+   * document body and provide them aswell.
+   */
+  private async getInlineStyleSheet(): Promise<string> {
+    const css = await FSE.readFile(
+      Path.join(__dirname, 'static', 'markdown.css'),
+      'utf8'
+    )
+
+    // scrape theme variables so iframe theme will match app
+    const docStyle = getComputedStyle(document.body)
+    const textColor = docStyle.getPropertyValue('--text-color')
+    const backgroundColor = docStyle.getPropertyValue('--background-color')
+    const codeBackgroundColor = docStyle.getPropertyValue(
+      '--box-alt-background-color'
+    )
+    const boxBorderColor = docStyle.getPropertyValue('--box-border-color')
+
+    return `<style>
+      :root {
+        --text-color: ${textColor};
+        --background-color: ${backgroundColor};
+        --code-background-color: ${codeBackgroundColor};
+        --box-border-color: ${boxBorderColor};
+      }
+      ${css}
+    </style>`
+  }
+
+  /**
+   * We still want to be able to navigate to links provided in the markdown.
+   * However, we want to intercept them an verify they are valid links first.
+   */
+  private setupLinkInterceptor(frameRef: HTMLIFrameElement): void {
+    frameRef.addEventListener('load', () => {
+      frameRef.contentDocument?.addEventListener('click', ev => {
+        const { contentWindow } = frameRef
+
+        if (contentWindow && ev.target instanceof contentWindow.Element) {
+          const a = ev.target.closest('a')
+          if (a !== null) {
+            ev.preventDefault()
+
+            if (/^https?:/.test(a.protocol)) {
+              this.props.onMarkdownLinkClicked?.(a.href)
+            }
+          }
+        }
+      })
+    })
+  }
+
+  /**
+   * Builds a <base> tag for cases where markdown has relative links
+   */
+  private getBaseTag(baseHref: string | null): string {
+    if (baseHref == null) {
+      return ''
+    }
+
+    const base = document.createElement('base', {})
+    base.href = baseHref
+    return base.outerHTML
+  }
+
+  /**
+   * Generates an iframe with a csp allowed script of a markdown parser.
+   */
+  private mountIframeContents = async (): Promise<void> => {
+    if (this.frameRef === null) {
+      return
+    }
+
+    const styleSheet = await this.getInlineStyleSheet()
+
+    const parsedMarkdown = marked(this.props.markdown, {
+      gfm: true,
+    })
+
+    const sanitizedHTML = DOMPurify.sanitize(parsedMarkdown)
+
+    const src = `
+      <html>
+        <head>
+          ${this.getBaseTag(this.props.baseHref)}
+        </head>
+        <body>
+          ${styleSheet}
+          <div id="content">${sanitizedHTML}</div>
```

Comment:
Do we need this to be encapsulated?

```suggestion
          ${sanitizedHTML}
```